source/extensions/retry/host/well_known_names.h

```diff
@@ -16,6 +16,7 @@ class RetryHostPredicatesNameValues {
 public:
   // Previous host predicate. Rejects hosts that have already been tried.
   const std::string PreviousHostsPredicate = "envoy.retry_host_predicates.previous_hosts";
+  const std::string OmitCanaryHostsPredicate = "community.retry_host_predicates.canary_host_omitting";
```

Comment:
Make the extension name `envoy.retry_host_predicates.omit_canary_hosts` in:

- [ ] Extension BUILD file
- [ ] Unit test BUILD file
- [ ] `extension_build_config.bzl`
- [ ] Documentation `docs/root/intro/arch_overview/http_connection_management.rst`